Add inline delivery tabs component

jfplataroti opened the pull request:
Create a new component `DeliveryTabsInline` that is pretty much the same as `DeliveryTabs` with two differences:
- Uses a `RadioButtonGroup` to represent the inline tabs
- fields are inline in the same row.

This component used in the `Layout` component when the screen size is not mobile.

<img width="1171" alt="tabs inline" src="https://user-images.githubusercontent.com/1827213/107064113-29010a00-67a9-11eb-9dbc-a6825d885070.png">

It still looks bad when you are using a small desktop screen (size medium) but not sure if we want to worry about that until we have the desktop mockups.

### Other code changes:
- unified localstorage keys for tabs and delivery fields into one.
- added tabs to `useDeliveryOptions`.

jfplataroti commented on packages/web/components/deliveryTabsInline.tsx:
```diff
@@ -0,0 +1,86 @@
+import theme from '../lib/theme';
+import { useTranslation } from '../i18n';
+
+const getTabOptionStyle = (option: TabOption) => {
+  return option.value === DeliveryMode.Delivery
+    ? { borderTopLeftRadius: theme.global.units.base }
+    : { borderTopRightRadius: theme.global.units.base };
+};
```
created this function to add the border-radius to the edges, base on the mode value. This assumes that delivery goes first and there are only two items. I tried to avoid iterating the tabs to check which one is at each edge.

@djoyal let me know if you think it's kind of brittle and we need to make it more robust.

djoyal replied:
The only other approach that might work would be to wrap the parent in the correct styling (E.G. try to apply the styling for the corners to the `RadioButtonGroup` instead of the children). I don't know if this would work though

djoyal commented on packages/web/components/deliveryTabsInline.tsx:
```diff
@@ -0,0 +1,86 @@
+        }}
+      </RadioButtonGroup>
+
+      {value.mode === DeliveryMode.Delivery && (
```
It looks like these two boxes could be combined, and the condition logic wrapped around the first component (EG: have one box, move the `value.mode === DeliveryMode...` around the first InputButton in the box (so that that's only 1 copy of the time picker, and the box chooses to display the address finder or the establishment picker)

jfplataroti replied:
oh yes, that would simplify the UI a bit more!

djoyal reviewed (approved):
This looks great! I agree that the format for the `medium` size display does not look great. I think we might also need to check the `Layout` component in another PR, it doesn't seem to be 100% response in some cases (I can get the `Sign In` button to disappear from the screen). That's a future issue though.

Nice work!

jfplataroti commented:
> The only other approach that might work would be to wrap the parent in the correct styling (E.G. try to apply the styling for the corners to the RadioButtonGroup instead of the children). I don't know if this would work though

Good idea, I'll give it a try.

djoyal reviewed (approved):
Thanks for the refactor!

jfplataroti commented:
Good call with the suggestions!

- removed the elements duplication, only wrapping with deliveryMode if the components `establishment` and `address` fields.
- removed the funky styling function to add border radius, now it's added with the styles of the `radioButtonGroup`.

djoyal reviewed (approved):
Nice job with the refactor, thanks!
